### DISCUSSION TITLE
Notifications tray

### DISCUSSION THREAD

**Discussion by petrochuk** [microsoft/powerplatform-build-tools#457] [Q&A]
Question to the community: If we integrate with Notifications tray what scenarios/short cuts/actions you would like to see?

![image](https://github.com/microsoft/powerplatform-build-tools/assets/30735471/6a637e2d-18d6-475c-9e96-fe87ced9483c)

**Comment by Laskewitz** [comment]
There are a couple of scenarios I can think of:

# Shortcuts
It could be cool to be able to click the icon in the notification tray and have a sort of browser in there that can show the environments in your auth profile and then be able to right click + go to maker portal or right click + go to admin portal.

It would be nice to favorite the environments as well, so that you can quickly go to your favorite environments

# Notifications
It would be nice to get notifications of long running operations like solution import, data import etc.

# Actions
This is a long shot, but it could be cool to be able to trigger scripts from there, that it would give a pop-up for user inputs and then run the script.

**Comment by appieschot** [comment]
I love the openness and discussion options, so let me start with kuddos for that! 🎉 Personally I am not really looking for notifications, I have turned of all notifications in Windows as it is. For long running operations I would be perfectly happy with a progress spinner in my terminal. 

For shortcuts I would see some added value, navigate to maker portal / navigate to active environment etc.. When working with the CLI I tend to use the terminal for everything so if those shortcuts could become commands that would help my productivity flow better then clicking an icon :-).

**Reply by Laskewitz** [reply]
I do believe if there is something like notifications, it should be easily configurable. Maybe as a user setting, so that you can easily switch notifications on or off.

**Reply by appieschot** [reply]
@laskewitz if notifications would be part of the notification tray they don't necessary require config, they listen to your windows preferences (system / notifications) that way, so a v1 with notifications wouldn't need to much config I would recon.

**Reply by Laskewitz** [reply]
@appieschot - agreed to having the quiet hours etc, but might be helpful to have a setting for which type of notifications you want; only on long running operations, or other categories...

**Comment by rajyraman** [comment]
+1 for @Laskewitz suggestion - toast notifications when long running operations finish.

Switching auth profiles when you have multiple auth profiles could be useful.
